python/pyarrow/serialization.py

```diff
@@ -55,6 +55,25 @@ def _deserialize_numpy_array_list(data):
         return np.array(data[0], dtype=np.dtype(data[1]))
 
 
+def _serialize_numpy_matrix(obj):
+    if obj.dtype.str != '|O':
+        # Make the array c_contiguous if necessary so that we can call change
+        # the view.
+        if not obj.flags.c_contiguous:
+            obj = np.ascontiguousarray(obj.A)
+        return obj.A.view('uint8'), obj.A.dtype.str
+    else:
+        return obj.A.tolist(), obj.A.dtype.str
+
+
+def _deserialize_numpy_matrix(data):
```

Comment:
I just did a little timing and the time taken by this method seems to scale linearly in the size of the array. Can we fix this?

I think `np.matrix(...)` is the slow line.

Reply:
Indeed, copying doesn't seem like the thing to do here. I think `np.matrix(data, copy=False)` would just create a new view.